### PR TITLE
feat/114-setup-vitepress

### LINKED ISSUE #114: Setup Vitepress
**Is your feature request related to a problem? Please describe.**
Documentation should be done using Vitepress SPA

**Describe the solution you'd like**
Initialize Vitepress project

**Describe alternatives you've considered**
---

**Additional context**
See an example: https://github.com/it-at-m/digiwf-core/tree/dev/docs

### PR DESCRIPTION
**Pull request by DanielOber**
**Description**

- added simple vitepress setup
- enabled search und mermaid
- added .gitignore
- added .npmrc

The Documentation and removing the examples should be done in separate tickets. Its just the first initialization that should be done here

**Reference**

Closes #114

### CODE REVIEW

File: docs/package.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "scripts": {
+    "docs:dev": "vitepress dev",
+    "docs:build": "vitepress build",
+    "docs:preview": "vitepress preview"
+  },
+  "devDependencies": {
+    "vitepress": "^1.0.0-rc.45",
```
Comment (simonhir) [inline docs/package.json:8]:
```suggestion
    "vitepress": "^1.3.1",
```

**Review by simonhir — APPROVED** [review]
LGTM except rc version

File: docs/.gitignore [hunk 2]
```diff
@@ -0,0 +1,191 @@
+# Covers Maven specific
```
Comment (simonhir) [inline docs/.gitignore:1]: I.e. Maven in the gitignore seems a bit odd or has this any purpose?

Comment (DanielOber) [reply docs/.gitignore:1]: its just our default .gitignore, that we have in every repository. We have #36 to combine them but we are still waiting for the webcomponent templates.

File: docs/package.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+  "scripts": {
+    "docs:dev": "vitepress dev",
+    "docs:build": "vitepress build",
+    "docs:preview": "vitepress preview"
+  },
+  "devDependencies": {
+    "vitepress": "^1.3.1",
+    "mermaid": "^10.9.0",
+    "vitepress-plugin-mermaid": "^2.0.16"
```
Comment (devtobi) [inline docs/package.json:10]: I would encourage the use of pinned dependencies here (as renovatebot will pin them later anyways)

**Review by devtobi — CHANGES_REQUESTED** [review]

**Review by simonhir — APPROVED** [review]

**Review by devtobi — APPROVED** [review]

**Pull request merged**